clang/lib/CIR/CodeGen/CIRGenModule.cpp

```diff
@@ -1267,8 +1267,14 @@ void CIRGenModule::emitTopLevelDecl(Decl *decl) {
     break;
 
   // No code generation needed.
-  case Decl::UsingShadow:
+  case Decl::ClassTemplate:
+  case Decl::Concept:
+  case Decl::CXXDeductionGuide:
   case Decl::Empty:
+  case Decl::FunctionTemplate:
+  case Decl::StaticAssert:
+  case Decl::TypeAliasTemplate:
```

Comment:
You can add `Decl::VarTemplate`, `Decl::VarTemplatePartialSpecialization`, `Decl::Block`, `Decl::Binding` to be complete and on par with classic codegen.